File: datastructures/symboltables/SequentialSearchST.java
public class SequentialSearchST<Key, Value> {

    private Node first;

    private class Node {
        Key key;
        Value val;
        Node next;
        public Node(Key key, Value val, Node next) {
            this.key = key;
            this.val = val;
            this.next = next;
        }
    }

    public Value get(Key key) {
        for (Node x = first; x != null; x = x.next) {
            if (key.equals(x.key)) return x.val;
        }
        return null;
    }

    public void put(Key key, Value val) {
        for (Node x = first; x != null; x = x.next) {
            if (key.equals(x.key)) {
                x.val = val;
                return;
            }
        }
        first = new Node(key, val, first);
    }

    public static void main(String[] args) {
        SequentialSearchST<String, Integer> st = new SequentialSearchST<>();
        st.put("a", 0);
        st.put("b", 1);
        st.put("c", 2);

        System.out.println("a: " + st.get("c"));
        System.out.println("a: " + st.get("b"));
        System.out.println("a: " + st.get("a"));
    }
}

File: datastructures/symboltables/SequentialSearchST.py
class Node:
    def __init__(self, key, val, next):
        self.key = key
        self.val = val
        self.next = next


class SequentialSearchST:
    first = None

    def __init__(self):
        pass

    def get(self, key):

        temp = self.first
        while temp:
            if temp.key is key:
                return temp.val

            # Advance the temp to the next
            temp = temp.next

        return -1

    def put(self, key, val):

        if not self.first:
            self.first = Node(key, val, None)
            return

        temp = self.first
        while temp:
            if temp.key is key:
                temp.val = val
            temp = temp.next

        self.first = Node(key, val, self.first)


if __name__ == '__main__':
    st = SequentialSearchST()

    st.put("a", 0)
    st.put("b", 1)
    st.put("c", 2)

    print(st.get("c"))
    print(st.get("b"))
    print(st.get("a"))

